File: BricksGDX/core/src/com/salvador/bricks/game_objects/MenuButton.java
package com.salvador.bricks.game_objects;

import static com.salvador.bricks.game_objects.Constants.BUTTON_EXIT;
import static com.salvador.bricks.game_objects.Constants.BUTTON_INFO;
import static com.salvador.bricks.game_objects.Constants.BUTTON_PLAY;
import static com.salvador.bricks.game_objects.Constants.BUTTON_RESET;
import static com.salvador.bricks.game_objects.ResourceManager.getTexture;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;

public class MenuButton extends Actor {

    public float x, y;
    public float w, h;
    public boolean touch;
    private Texture menuButton;

    public MenuButton(int type, float x, float y, float w, float h) {
        this.x = x;
        this.y = y;
        this.w = w;
        this.h = h;

        switch (type) {
            case BUTTON_PLAY:
                menuButton = getTexture("button.png");
                break;
            case BUTTON_EXIT:
                menuButton = getTexture("btn_exit.png");
                break;
            case BUTTON_INFO:
                menuButton = getTexture("btn_info.png");
                break;
            case BUTTON_RESET:
                menuButton = getTexture("btn_reset.png");
                break;
        }
        setBounds(x, y, w, h);

        addListener(new InputListener() {
            @Override
            public boolean touchDown(InputEvent event, float x, float y,
                                     int pointer, int button) {
                touch = true;
                return true;
            }

            public void touchDragged(InputEvent event, float x, float y, int pointer) {
                super.touchDragged(event, x, y, pointer);
            }
        });
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        batch.draw(menuButton, x, y, w, h);
    }
}

File: BricksGDX/core/src/com/salvador/bricks/game_objects/MenuButton.kt
package com.salvador.bricks.game_objects

import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener
import com.salvador.bricks.game_objects.Constants.BUTTON_EXIT
import com.salvador.bricks.game_objects.Constants.BUTTON_INFO
import com.salvador.bricks.game_objects.Constants.BUTTON_PLAY
import com.salvador.bricks.game_objects.Constants.BUTTON_RESET

class MenuButton(type: Int, var x: Float, var y: Float, var w: Float, var h: Float) : Actor() {

    @JvmField
    var touch = false
    private var menuButton = when (type) {
        BUTTON_PLAY -> ResourceManager.getTexture("button.png")
        BUTTON_EXIT -> ResourceManager.getTexture("btn_exit.png")
        BUTTON_INFO -> ResourceManager.getTexture("btn_info.png")
        BUTTON_RESET -> ResourceManager.getTexture("btn_reset.png")
        else -> {
            null
        }
    }

    init {
        setBounds(x, y, w, h)

        addListener(object : InputListener() {
            override fun touchDown(
                event: InputEvent, x: Float, y: Float,
                pointer: Int, button: Int
            ): Boolean {
                touch = true
                return true
            }

        })
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        batch.draw(menuButton, x, y, w, h)
    }
}
